RestrictEvents/SoftwareUpdate.cpp

```diff
@@ -200,15 +200,19 @@ static int my_sysctl_vmm_present(__unused struct sysctl_oid *oidp, __unused void
     char procname[64];
     proc_name(proc_pid(req->p), procname, sizeof(procname));
     // SYSLOG("supd", "\n\n\n\nsoftwareupdated vmm_present %d - >>> %s <<<<\n\n\n\n", arg2, procname);
-    if (strcmp(procname, "softwareupdated") == 0 || strcmp(procname, "com.apple.Mobile") == 0) {
-        int hv_vmm_present = 1;
-        return SYSCTL_OUT(req, &hv_vmm_present, sizeof(hv_vmm_present));
+    if (checkKernelArgument("-revsbvmm") && (strcmp(procname, "softwareupdated") == 0 || strcmp(procname, "com.apple.Mobile") == 0)) {
+		int hv_vmm_present_on = 1;
+		return SYSCTL_OUT(req, &hv_vmm_present_on, sizeof(hv_vmm_present_on));
     }
+	else if (checkKernelArgument("-revasset") && (strcmp(procname, "AssetCache") == 0 || strcmp(procname, "AssetCacheManage") == 0)) {
```

Comment:
Firstly compare against Asset with strncmp, then check for the suffix to reduce unnecessary comparisons.

Reply:
Adjusted with `strncmp`, however omitted suffix check as other AssetCache* processes should be spoofed as well.